src/core.c/CompUnit/PrecompilationStore.pm6

```diff
@@ -3,6 +3,10 @@ role CompUnit::PrecompilationStore {
     method new-unit(| --> CompUnit::PrecompilationUnit:D)
     { ... }
 
+    # Initiates the item-level lock
+    method initiate-lock(IO::Path:D $path)
```

Comment:
Just because one of the implementations of this interface is a `::File` doesn't mean they all are. Enforcing `IO::Path` here seems to be hard coupling an interface to a specific implementation.